src/terrain/index.js

```diff
@@ -0,0 +1,157 @@
+    return mockArray;
+};
+
+createMockArray = (boardBounds, terrainBounds) => {
+    const boardX = boardBounds.x;
+    const boardY = boardBounds.y;
+    const terrainX = terrainBounds.x;
+    const terrainY = terrainBounds.y;
+
+    const mockArray = [];
+    const mockWidth =  boardX/terrainX;
+    const mockHeight = boardY/terrainY;
+
+    const defaultInfo = {
+        filled: false,
+        north: true,
+        south: true,
+        east: true,
+        west: true
+    };
+
+    for(let i = 0; i < mockWidth; i++) {
+        mockArray[i] = [];
+        for (let j = 0; j < mockHeight; j++) {
+            const info = { ...defaultInfo };
+            if (i === 0) {
+                info.west = false;
+            } else if (i === mockWidth - 1) {
+                info.east = false;
+            }
+            if (j === 0) {
+                info.north = false;
+            } else if (j === mockHeight - 1) {
+                info.south = false;
+            }
+            mockArray[i].push(info);
+        }
+    }
+    return {
+        mockArray,
+        mockWidth,
+        mockHeight
+    };
+};
+
+const deepClone = (toCopy) => {
+    const toReturn = [];
+    toCopy.forEach((row, rowNum) => {
+        toReturn[rowNum] = [];
+        row.forEach(elem => {
+            toReturn[rowNum].push({ ...elem });
+        });
+    });
+    return toReturn;
+};
+
+
+const cleanUpBoard = (board) => {
+    for (let i = 0; i < board.length; i++) {
+        for (let j = 0; j < board[i].length; j++) {
+            board[i][j] = {
+                filled: board[i][j].filled,
+                north: board[i][j].north,
+                south: board[i][j].south,
+                east: board[i][j].east,
+                west: board[i][j].west
+            };
+        }
+    }
+};
+
+const verifyKeyPoint = (board, keyPoint, boardWidth, boardHeight) => {
+    const pointsVerified = [];
+    for (let x = 0; x < boardWidth; x++) {
+        for (let y = 0; y < boardHeight; y++) {
+            let test = { x, y };
+            if ((pointsVerified.findIndex(point => point.x === test.x && point.y === test.y) > -1) ||
+                (test.x === keyPoint.x && test.y === keyPoint.y) ||
+                (board[test.x][test.y].filled)
+            ) {
+                continue;
+            }
+            const pointsOnCurrentPath = [];
+            const currentPoint = { ...test };
+            while(true) {
+                const boardPoint = board[currentPoint.x][currentPoint.y];
+                if (
+                    (pointsVerified.findIndex(point => point.x === currentPoint.x && point.y === currentPoint.y) > -1) ||
+                    (currentPoint.x === keyPoint.x && currentPoint.y === keyPoint.y)
+                ) {
+                    pointsVerified.push(...pointsOnCurrentPath);
+                    break;
+                } else if(
+                    (!boardPoint.north || boardPoint.wentNorth) &&
```

Comment:
It's around here where my eyes + brain said "nah". I think it would be helpful to break this up or maybe pull out some of the "rules" of the algorithm to separate them from this condition in a loop in a loop in a loop

Reply:
yeah I went ahead and cleaned up a lil bit and included some comments to hopefully clear up what I was trying to get at, can you let me know what you think?